### PR TITLE
IGNITE-20417 Get rid of unused code in the index module

### PR DESCRIPTION
**Pull request by tkalkirill**
https://issues.apache.org/jira/browse/IGNITE-20417

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/index/ItIndexManagerTest.java [hunk 1]
```diff
@@ -53,6 +53,8 @@ public void eventsAreFiredWhenIndexesCreatedAndDropped() {
 
         String tableName = "TNAME";
 
+        int catalogVersion = ((IgniteImpl) ignite).catalogManager().latestCatalogVersion();
```
[the remark this answers is not in the document]
Comment (tkalkirill) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/index/ItIndexManagerTest.java:56]: fix it

File: modules/index/src/main/java/org/apache/ignite/internal/index/IndexManager.java [hunk 2]
```diff
@@ -215,22 +217,21 @@ private CompletableFuture<Boolean> onIndexCreate(CreateIndexEventParameters para
...
     private CompletableFuture<Void> createIndexLocally(
             long causalityToken,
+            int catalogVersion,
             CatalogTableDescriptor table,
             CatalogIndexDescriptor index
     ) {
         int tableId = table.id();
         int indexId = index.id();
 
-        if (LOG.isInfoEnabled()) {
```
[the remark this answers is not in the document]
Comment (tkalkirill) [reply modules/index/src/main/java/org/apache/ignite/internal/index/IndexManager.java:224]: This is not a hot path, I just find it more beautiful this way, checking happens inside.

[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/index/src/main/java/org/apache/ignite/internal/index/IndexManager.java:224]: we have a lot of these checks all over the code, I'd prefer to keep the code consistent. It is also not related, so please revert it back

**Review by sashapolo — APPROVED** [review]

File: modules/index/src/main/java/org/apache/ignite/internal/index/IndexManager.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (tkalkirill) [reply modules/index/src/main/java/org/apache/ignite/internal/index/IndexManager.java:224]: fix it

**Review by sashapolo — APPROVED** [review]

**Pull request merged**